src/recovery/impl/thread_local_logger.cpp

```diff
@@ -36,7 +36,10 @@
 namespace LineairDB {
 namespace Recovery {
 
-ThreadLocalLogger::ThreadLocalLogger() { LineairDB::Util::SetUpSPDLog(); }
+ThreadLocalLogger::ThreadLocalLogger(const Config& config)
+  : WorkingDir(config.lineairdb_logs_dir) {
+    LineairDB::Util::SetUpSPDLog();
```

Comment:
Let me clarify the unnecessary thing.

if you mean `SetUpSPDLog()` call here is unnecessary, I'm not sure that because the call exists before. I didn't change the logic in this PR.
if you mean `WorkingDir` variable initialization and passing of config reference are unnecessary, IMO, those are required on GetLogFileName and GetWorkingLogFileName methods to concatenate the path.

Or, I may be missing something sorry.

Reply:
> I'm not sure that because the call exists before.

I'm sorry, I misunderstood that the call didn't exist before. I apologize for that.